feat(filtering): Open + Download Job's Trace logs

azachar opened the pull request:
Hi there, 

this is a WIP PR preview of trace log downloading and filtering. 
I have added a dialog and some fancy features to operate on a daily basis with GitLab's logs.
So if you run some protractor tests that you need to examine, this should help a lot.

Here is a list of some features:
**opens**: GitLab page for pipelines, pipeline or job
**downloads**: trace log, all failed trace logs
**filter**: based on your domain package your failed stack traces (you need to install additional package for enabling this)
**editor**: support opening log with an ANSI editor that allows custom filtering
**key bindings:** to open GitLab's pipelines page and the job selector (ctrl+alt+g) and (shift+ctrl+alt+g)

Unfortunately,  download progress is not shown yet. But perhaps some nice contributors could extend this and polish. Also, tests are missing. 

Currently, some features are borrowed from additional 2 packages that need to be installed, otherwise filtering will not work that nicely. All is in mentioned in readme.

Hope to hear some feedback from you.

Cheers,
Andrej

azachar committed:
feat(filtering): Open + Download Job's Trace logs

opens: GitLab page for pipelines, pipeline or job
downloads: trace log, all failed trace logs
editor: support opening log with an ANSI editor that allows custom filtering
key bindings: to open GitLab's pipelines page and the job selector

azachar committed:
doc(filtering): how to install additional packages

azachar committed:
feat(filtering): reload

azachar committed:
fix(filtering): npe in job selector

blakawk reviewed (changes requested):
Please split your PR in several smaller ones and avoid referencing your packages directly. An user installing gitlab-integration should be able to use it without cloning from github.

azachar committed:
feat(pipelines): pipeline selector

azachar committed:
fix(update): update command

azachar committed:
style(job): nicer job selector

azachar committed:
fix(pipeline): clean up

azachar committed:
fix(view): double initialization

azachar committed:
feat(job): show always failed jobs, then success and last unstable

azachar committed:
fix(pipeline): clean up

azachar committed:
feat(progress): color based on success time

azachar committed:
feat(job): selector filtering

azachar committed:
feat(pipeline): sort actions

fix(pipeline): npe

azachar committed:
feat(job): commit message header

azachar committed:
style(selectors): commit title instead of message

azachar committed:
fix(pipeline): search

azachar committed:
fix(pipeline): sorting

azachar committed:
feat(job): selector jobs sorting

azachar committed:
style(pipeline): sha info

azachar committed:
refactor(avatar): moved from job to pipeline

azachar committed:
fix(pipeline): skipped status

azachar committed:
feat(pipeline): abs elapsed time

azachar committed:
feat(pipeline): relative time

azachar committed:
fix(pipeline): always fails duplicated

azachar committed:
fix(pipeline): undefined if no jobs

blakawk reviewed (commented):
Given the size of this merge request, I did not have time to review it all. Please be patient.

azachar committed:
feat(job): sort by duration

azachar committed:
fix(progress): initial progress bar is calculate for jobs and pipeline

azachar committed:
feat(pipeline): sort by duration

azachar committed:
feat(all): pipeline selector for all pipelines regardless branch

feat(all): filter success, unstable and failed pipelines
fix(gitlab): update pipeline selected by user

azachar commented:
Well, It really becomes an epic, I agree :). So take your time. As being said it is a preview, an experiment. I am just testing what suites my needs perhaps you can pick it up and bring it to the next level. I am sure that features like caching and data model needs to be completely rewritten. But for now it "works for me" and it already brings a lot of things such us pipeline selector for all pipelines, statistics to find out which jobs are problematic and so on...  Have a look yourself :)

azachar committed:
fix(search): search keywords are based on template id: status: name: and so on
